code/model/modelread.cpp

```diff
@@ -2592,20 +2590,35 @@ int read_model_file(polymodel * pm, const char *filename, int n_subsystems, mode
 					}
 				}
 
+				// certain old models specify the submodel number, so let's maintain compatibilty
+				if (submodel_name != nullptr && can_construe_as_integer(submodel_name)) {
+					pm->submodel[i].look_at_submodel = atoi(submodel_name);
+					submodel_name = nullptr;
+				}
+
 				// did we fail to find it?
 				if (submodel_name != nullptr) {
 					Warning(LOCATION, "Unable to match %s %s $look_at: target %s with a submodel!\n", pm->filename, pm->submodel[i].name, submodel_name);
 					pm->submodel[i].look_at_submodel = -1;
+					pm->submodel[i].movement_type = MOVEMENT_TYPE_NONE;
 				}
 				// are we navel-gazing?
 				else if (pm->submodel[i].look_at_submodel == i) {
 					Warning(LOCATION, "Matched %s %s $look_at: target with its own submodel!  Submodel cannot look at itself!\n", pm->filename, pm->submodel[i].name);
 					pm->submodel[i].look_at_submodel = -1;
+					pm->submodel[i].movement_type = MOVEMENT_TYPE_NONE;
 				}
 			}
 		}
 	}
 
+	// And now look through all the submodels and set the model flag if any are intrinsic-rotating
+	for (i = 0; i < pm->n_models; i++) {
+		if (pm->submodel[i].movement_type == MOVEMENT_TYPE_INTRINSIC_ROTATE) {
+			pm->flags |= PM_FLAG_HAS_INTRINSIC_ROTATE;
```

Comment:
Can't we break here? It looks like we only have to find one submodel with the flag.

Reply:
That's a good idea.

Reply:
That's a good idea.